### DISCUSSION TITLE
Want example of Fobi Forms accessed through Django REST Framework

### DISCUSSION THREAD

**Discussion by trexmo** [barseghyanartur/django-fobi#291] [Q&A]
I have Fobi forms working fine. Now I want to access the forms via Django REST Framework but cannot find documentation or examples.  The doces say they are integrated but that is about it.

Thanks for your help.

**Comment by trexmo** [comment]
I saw in another post in the Django Forum that you cannot POST to a Fobi form, but you would need to POST to a view and then apparently write some code to populate the Fobi form.    

I am confused, when you enter data into a Fobi form and submit it, I see in the terminal that it is a POST.  For example I created a Fobi form and from the Dashboard entered data into that form and submitted.  First I see a GET of **/fobi/view/sample-_raining/**

Then when I submit I see in the terminal, **POST /fobi/view/sample-_raining/ HTTP/1.1" 302 0** and the 302 forwards to **/fobi/view/sample-_raining/submitted/** which returns the "Success" page.

So, why cannot I use PostMan or my own code to do that POST, and if it is easier to create my own View and use the DRF to post to that and then use python to populate the form, how would that be done?

**Reply by trexmo** [reply]
So the CSRF mechanism may overcomplicate trying to post to a Fobi form, and I can use a DRF view to post to and use code to populate the form...but found no documentation on how to populate a form in my code.

**Comment by trexmo** [comment]
I was able to populate my form with DRF using put 

**http://localhost:8000/api/fobi-form-entry/sample_training/** using Postman with a raw body of JSON

```
{
    "param1": "Hello",
    "param2": "World from Postman",
    "description": "now is the time for all good men to come to the aid of their country.",
    "notify": "mo@globalsaassol.com"
}
```
It is a PUT not a POST which was my mistake looking through the docs for POST.

**Comment by barseghyanartur** [comment]
In most of the cases, if docs are not available, make sure to check the correspondent tests.

https://github.com/barseghyanartur/django-fobi/blob/main/src/fobi/tests/test_drf_integration.py
